sendgrid/base_interface.py

```diff
@@ -60,3 +63,31 @@ def send(self, message):
             message = message.get()
 
         return self.client.mail.send.post(request_body=message)
+
+    def set_host(self,host):
+        self.host = host
+        self.client = python_http_client.Client(
+            host=self.host,
+            request_headers=self._default_headers,
+            version=3)
+
+    def set_data_residency(self,region):
+        """
+        * Client libraries contain setters for specifying region/edge.
+        * This allows support global and eu regions only. This set will likely expand in the future.
+        * Global should be the default
+        * Global region means the message should be sent through:
+        * HTTP: api.sendgrid.com
+        * EU region means the message should be sent through:
+        * HTTP: api.eu.sendgrid.com
+        :param region:
+        :return:
+        """
```

Comment:
I don't think the `*` are necessary here. Also missing the parameter description. You can refer to the other existing function definitions for an example